ZichanYang324/BF-Employee-HR
# BGP-10 [Frontend and Backend] Personal Information Page

# Description
Implements # (BGP-10) 

## Related PRS (if any):
This frontend PR is related to the #20  backend PR.

## Main changes explained:
- Created infoSlice to store
- Created `/info` page
- Created corresponding apis under the `/info` route.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` on both FE and BE to run this PR locally
3. To get data, run seed `npm run seed` (run it on a safe db, as it will reset collections)
4. Make sure a user is newly logged in. Go to /info page.
5. Except for the first **Name** section, all other section should update the DB correctly.

## Screenshots or videos of changes or expecting result:
![Screenshot 2024-02-27 at 8 07 02 PM](https://github.com/ZichanYang324/BF-Employee-HR/assets/111475191/d60a61e6-b1d6-4877-a1a6-879c4d527019)
![Screenshot 2024-02-27 at 8 07 12 PM](https://github.com/ZichanYang324/BF-Employee-HR/assets/111475191/b100df20-630f-4624-8fa3-4cf02c51d049)



## Note:

### TODOs:

1. Finish the api for updating **Name** section
2. Add user input validation
